src/cljs/harja/tiedot/urakka/laadunseuranta/talvihoitoreitit_tiedot.cljs

```diff
@@ -0,0 +1,230 @@
+(defonce valitut-kohteet-atom (r/atom #{}))
+
+(defonce karttataso-talvihoitoreitit (r/atom []))
+(defonce karttataso-nakyvissa? (r/atom true))
+
+(defonce talvihoitoreitit-kartalla
+  (reaction
+    (let [;; Näytä vain valittu kohde kartalla
+          valitut-kohteet (if (= (count @valitut-kohteet-atom) 0)
+                            #{}
+                            @valitut-kohteet-atom)
+          talvihoitoreitit (keep (fn [reitti] (when (contains? valitut-kohteet (:id reitti)) reitti)) @karttataso-talvihoitoreitit)
+          sijainnit (map (fn [rivi] (map :sijainti (:reitit rivi))) talvihoitoreitit)
+          extentit (harja.geo/extent-monelle (flatten sijainnit))
+          map-reitit (into [] (flatten (mapv (fn [reitti]
+                                               (mapv (fn [r]
+                                                       (when (:sijainti r)
+                                                         {:alue (merge {:tyyppi-kartalla :talvihoitoreitit
+                                                                        :stroke {:width 8
+                                                                                 :color (:vari reitti)}}
+                                                                  (:sijainti r))
+                                                          :extent (harja.geo/extent (:sijainti r))
+                                                          :tyyppi-kartalla :talvihoitoreitit
+                                                          :selite {:teksti "Talvihoitoreitti"
+                                                                   :img (pinni-ikoni "sininen")}
+                                                          :infopaneelin-tiedot {:pituus (:pituus r)
+                                                                                :nimi (:nimi reitti)}
+                                                          :ikonit [{:tyyppi :merkki
+                                                                    :paikka [:loppu]
+                                                                    :zindex 21
+                                                                    :img (pinni-ikoni "sininen")}]}))
+                                                 (:reitit reitti)))
+                                         talvihoitoreitit)))]
+      (when (and (not-empty talvihoitoreitit) (not (nil? talvihoitoreitit)) @karttataso-nakyvissa?)
+        (with-meta map-reitit
+          {:selitteet [{:vari (map :color asioiden-ulkoasu/talvihoitoreitit)
+                        :teksti "Talvihoitoreitit"
+                        :extent extentit}]
+           :extent extentit})))))
+
+;; Talvihoitoreittien excel-tiedoston lataamisessa käyttäjän informointiin tarkoitettu modal
+(defn tulos-modal [leikkaavat-virheet muut-virheet paivitetyt onnistuneet otsikko]
+  (let [otsikkotyyli (if (or (not (empty? onnistuneet)) (not (empty? paivitetyt))) :onnistunut :virhe)]
```

Comment:
Öh, sori kun viilailen vähän pilkkua mutta not empty kahdesti.  Tässä olisi parempi tapa jotain 

`(not (and (empty? onnistuneet) (empty? paivitetyt)))`

Tai vaikka 

`(not (every? empty? [onnistuneet paivitetyt]))`

Reply:
Mä tykkään tuosta not emptystä. Helppo lukea, kaikki ymmärtää heti, mistä on kyse. Mutta clojure ehdottaa itse, että käytetään seq funkkaria, not emptyn tilalta, niin vaihdoin siihen.